src/commands/cmd_server.cc

```diff
@@ -1123,6 +1126,75 @@ class CommandRdb : public Commander {
   uint32_t db_index_ = 0;
 };
 
+class CommandAnalyze : public Commander {
+ public:
+  Status Parse(const std::vector<std::string> &args) {
+    if (args.size() <= 1) return {Status::RedisExecErr, errInvalidSyntax};
+    for (int i = 1; i < args.size(); ++i) {
+      true_args_.push_back(args[i]);
+    }
+    return Status::OK();
+  }
+  Status Execute(Server *srv, Connection *conn, std::string *output) {
+    auto commands = redis::CommandTable::Get();
+    auto cmd_iter = commands->find(util::ToLower(true_args_[0]));
+    if (cmd_iter == commands->end()) {
+      // unsupported redis command
+      return {Status::RedisExecErr, errInvalidSyntax};
+    }
+    auto redis_cmd = cmd_iter->second;
+    auto cmd = redis_cmd->factory();
+    cmd->SetAttributes(redis_cmd);
+    cmd->SetArgs(true_args_);
+
+    auto s = cmd->Parse(true_args_);
+    if (!s.IsOK()) {
+      return s;
+    }
+
+    auto prev_perf_level = rocksdb::GetPerfLevel();
+    rocksdb::SetPerfLevel(rocksdb::PerfLevel::kEnableTimeExceptForMutex);
+    rocksdb::get_perf_context()->Reset();
+    rocksdb::get_iostats_context()->Reset();
+
+    std::string tmp_output;
+    s = cmd->Execute(srv, conn, &tmp_output);
+    if (!s.IsOK()) {
+      return s;
+    }
+
+    if (tmp_output[0] == '-') {
+      *output = tmp_output;
+      return s;
+    }
+
+    std::string perf_context = rocksdb::get_perf_context()->ToString(true);
+    std::string iostats_context = rocksdb::get_iostats_context()->ToString(true);
+    rocksdb::get_perf_context()->Reset();
+    rocksdb::get_iostats_context()->Reset();
+    rocksdb::SetPerfLevel(prev_perf_level);
+    if (perf_context.empty()) {
+      *output = tmp_output;
+      return Status::OK();  // request without db operation
+    }
+
+    if (tmp_output.at(0) == '*') {
+      // if multi bulkstring, append directly
+      tmp_output.append(redis::BulkString(perf_context));
+      tmp_output.append(redis::BulkString(iostats_context));
+      *output = tmp_output;
+    } else {
+      tmp_output = redis::ExtractData(tmp_output);
+      *output = redis::MultiBulkString(
+          {std::move(tmp_output), "(perf_context): " + perf_context, "(iostats_context): " + iostats_context});
+    }
```

Comment:
```suggestion
    *output = redis::MultiLen(3); // command output + perf context + iostats context
    *output += std::move(tmp_output);
    *output += redis::BulkString(perf_context);
    *output += redis::BulkString(iostats_context);
```

This implementation isn't correct. For the multi-bulk strings, the first element is the number of array elements, so you can't simply append the element.

Reply:
Can also remove function `ExtractData` and rename `tmp_output` to `command_output`

Reply:
What do you mean by saying remove `ExtractData` function?

Reply:
Sounds good I've changed them accordingly :)